bot/exts/events/advent_of_code/_cog.py

```diff
@@ -52,13 +60,74 @@ def __init__(self, bot: Bot):
         self.status_task.set_name("AoC Status Countdown")
         self.status_task.add_done_callback(_helpers.background_task_callback)
 
+        self.completer_task.start()
+
+    @tasks.loop(minutes=10.0)
+    async def completer_task(self) -> None:
+        """
+        Give members who have completed all 50 AoC stars completionist role.
+
+        Runs on a schedule, as defined in the task.loop decorator.
+        """
+        await self.bot.wait_until_guild_available()
+        guild = self.bot.get_guild(Client.guild)
+        completionist_role = guild.get_role(Roles.aoc_completionist)
+        if completionist_role is None:
+            log.warning("Could not find the AoC completionist role, cancelling sync task.")
+            self.completer_task.cancel()
+            return
+
+        aoc_name_to_member_id = {
+            aoc_name: member_id
+            for member_id, aoc_name in await self.account_links.items()
+        }
+
+        try:
+            leaderboard = await _helpers.fetch_leaderboard()
+        except _helpers.FetchingLeaderboardFailedError:
+            await self.bot.send_log("Unable to fetch AoC leaderboard during role sync.")
+            return
+
+        placement_leaderboard = json.loads(leaderboard["placement_leaderboard"])
+
+        for member_aoc_info in placement_leaderboard.values():
+            if not member_aoc_info["stars"] == 50:
+                # Only give the role to people who have completed all 50 stars
+                continue
+
+            member_id = aoc_name_to_member_id.get(member_aoc_info["name"], None)
+            if not member_id:
+                continue
+
+            member = await members.get_or_fetch_member(guild, member_id)
+            if member is None:
+                continue
+
+            if completionist_role in member.roles:
+                continue
+
+            await members.handle_role_change(member, member.add_roles, completionist_role)
```

Comment:
So you don't want to remove the role from users that have left the leaderboard?

Reply:
I don't think that's all too important. If the user wants their role removed they can ask and we can block them. 

The role itself will only last for a month after AoC anyway.